app/safe/internal/server/route/receive.go

```diff
@@ -65,6 +67,12 @@ func ReceiveKeys(cid string, w http.ResponseWriter, r *http.Request, spiffeid st
 	keysCombined := agePrivateKey + "\n" + agePublicKey + "\n" + aesCipherKey
 	state.SetRootKey(keysCombined)
 
+	if env.RootKeyInputModeManual() && env.ManualRootKeyUpdatesKubernetesSecret() {
+		if err := bootstrap.PersistKeys(agePrivateKey, agePublicKey, aesCipherKey); err != nil {
+			log.FatalLn(&cid, "Failed to persist keys", err.Error())
```

Comment:
Let’s not crash, but just warn or something.

My worry is: if safe is not working with a backing store, killing it will lose whatever is in its memory.

`log.FatalLn` kills the process by design. — instead, the user might want to retry with a different key, or at a different time maybe.